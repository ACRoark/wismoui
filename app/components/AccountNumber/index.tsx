import * as React from 'react';

import formatAccountNumber from './formatAccountNumber';

interface IAccountNumberProps {
  value: string;
}

const AccountNumber: React.FC<IAccountNumberProps> = (props: IAccountNumberProps): React.ReactElement => {
  const { value } = props;

  const formattedAccountNumber = formatAccountNumber(value);

  return <span>{formattedAccountNumber}</span>;
};

export default AccountNumber;
