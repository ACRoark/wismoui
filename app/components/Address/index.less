@import (reference) '../../styles/themes/index.less';

@address-cls-prefix: ~'@{dte-prefix}-address';

.@{address-cls-prefix} {
  display: inline;
  font-style: normal;
  line-height: 1.375rem;
  white-space: pre-line;
}
