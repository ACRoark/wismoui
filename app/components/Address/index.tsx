import React from 'react';

import AddressFormat from 'components/Address/AddressFormat';
import IAddress from 'types/IAddress';

import formatAddress from './formatAddress';

import './index.less';

interface IAddressProps {
  address: IAddress;
  format?: AddressFormat;
}

const Address: React.FC<IAddressProps> = (props: IAddressProps): React.ReactElement => {
  const { address, format = 'block' } = props;

  return <address className="dte-wismo-address">{formatAddress(address, format)}</address>;
};

export default Address;
