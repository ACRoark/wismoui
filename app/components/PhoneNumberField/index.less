@import (reference) '../../styles/themes/index.less';

@phone-number-edit-cls-prefix: ~'@{dte-prefix}-phone-number-field';

.@{phone-number-edit-cls-prefix} {
  display: flex;
  flex-direction: row;

  &-value {
    font-size: @dte-wismo-font-size-0;
  }
}
