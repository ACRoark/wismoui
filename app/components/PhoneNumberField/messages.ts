import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.phone-number-field';

export default defineMessages({
  label: {
    id: `${scope}.label`,
    defaultMessage: 'Contact Number',
  },
});
