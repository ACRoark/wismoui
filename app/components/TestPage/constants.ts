import IOrderRequest from 'types/IOrderRequest';
import IProduct from '../../types/IProduct';

export const fakeProductList: IProduct[] = [
  {
    productType: 'GCR_A_H',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1_1',
    serviceDate: '2020-04-22',
  },
];

export const fakeServiceRequest: IOrderRequest = {
  accountNumber: '111122233334',
  address: {
    city: 'Bloomfield',
    line1: '123 Main',
    line2: null,
    state: 'MI',
    zip: '48301',
  },
  contactPhoneNumber: '7348675309',
  orderRequestStatusUpdates: [
    {
      createdAtDate: '20200319',
      createdAtTime: '131530',
      orderRequestStatus: 'REQUESTED',
    },
  ],
  orderRequestType: 'MIMO_START',
  premiseId: 'premise123',
  products: [
    {
      productType: 'GCR_A_H',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1_1',
      serviceDate: '2020-04-22',
    },
  ],
  serviceOrderEvents: [
    {
      appointment: {
        date: '2020-04-17',
        slotType: 'AM',
      },
      latestStatus: {
          createdAtDate: '20200305',
          createdAtTime: '223443',
          serviceOrderEventStatus: 'COMPLETED',
        },
        product: 'EFR_D1',
      serviceOrderCategory: null,
      serviceOrderId: '123456789',
    },
  ],
  verificationEvents: [
    {
      verificationCategory: 'R975',
      verificationEventId: '123123',
      verificationEventStatusUpdates: [
        {
          verificationCode: 'AD',
          verificationEventStatus: 'COMPLETED',
          createdAtDate: '20200319',
          createdAtTime: '131530',
        },
      ],
    },
  ],
  wantDate: '2020-04-22',
};

export const fakeStartServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_START',
};

export const fakeStopServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_STOP',
};
