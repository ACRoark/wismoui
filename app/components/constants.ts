import IApplicationConfig from 'types/IApplicationConfig';

export const PageTitles = Object.freeze({
  ORDER_STATUS_PAGE: 'Track Your Order | DTE Energy',
  SEARCH_PAGE: 'Order Tracker | DTE Energy',
  SUMMARY_PAGE: 'Track Your Order | DTE Energy',
});

export const testConfig: IApplicationConfig = {
  flags: {
    canChangeLanguage: false,
    developerMode: false,
  },
  gigyaSiteKey: 'foo',
  googleTagManagerConfig: {
    auth: 'p3zyt1YCX84XB82sOrjMGw', // '',
    id: 'GTM-K2WP588', // 'GTM-PBB6QZ4',
    preview: 'env-1', // '',
  },
  loading: true,
  recaptchaSitekey: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI',
  urls: {
    assistanceURL: 'https://www.dteenergy.com',
    backToDTE: 'https://www.dteenergy.com',
    backToMIMO: 'https://www.dteenergy.com',
    contactUs: 'https://www.contact.us.com',
    createAccount: 'https://www.create.account.com',
    decryptTokenApi: 'https://web-test.dteenergy.com/api/decryptToken?detokenizeValue=',
    documentSubmission: 'https://web-test.dteenergy.com/wps/wcm/connect/dte-web/quicklinks/web-form?detokenizeValue=',
    dynatraceScript:
      'https://dynatrace.dteenergy.com:9999/jstag/managed/98834406-1ae1-4257-992c-5b3c5f828584/968d8c7df07fe18c_complete.js',
    editPhoneNumber: 'https://www.edit.phone.number.com',
    editWantDate: 'https://www.edit.service.date.com',
    encryptTokenApi: 'https://web-test.dteenergy.com/api/encryptToken',
    findOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}?name={name}',
    getOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}',
    guestPay: 'https://www.guest.pay.com',
    headerLogo: 'https://www.header.logo.com',
    privacyPolicy: 'https://www.privacy.policy.com',
    searchOrders: 'https://firebase.or.jsonserver/orders',
    signIn: 'https://www.sign.in.com',
    termsAndConditions: 'https://www.terms.and.conditions.com',
  },
};
