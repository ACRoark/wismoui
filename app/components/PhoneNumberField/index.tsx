import React from 'react';
import 'react-responsive-modal/styles.css';

import BaseFormField from 'components/BaseField';
import EditPhoneNumberLink from 'components/EditPhoneNumberLink';
import PhoneNumber from '../PhoneNumber';

import messages from './messages';

import './index.less';

interface IPhoneNumberFieldProps {
  phoneNumber: string | null;
  serviceDate: string;
}

const PhoneNumberField: React.FC<IPhoneNumberFieldProps> = (props: IPhoneNumberFieldProps): React.ReactElement => {
  const { phoneNumber, serviceDate } = props;

  return (
    <>
      <BaseFormField label={messages.label}>
        <div className="dte-wismo-phone-number-field">
          <PhoneNumber className="dte-wismo-phone-number-field-value" value={phoneNumber} />
          {phoneNumber && <EditPhoneNumberLink serviceDate={serviceDate} />}
        </div>
      </BaseFormField>
    </>
  );
};

export default PhoneNumberField;
