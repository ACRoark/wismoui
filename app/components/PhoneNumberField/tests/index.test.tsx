import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import { testConfig } from 'components/constants';
import useConfig from 'hooks/useConfig';
import useFlags from 'hooks/useFlags';
import useModal from 'hooks/useModal';
import useThunkDispatch from 'hooks/useThunkDispatch';
import { userIsAuthenticated } from 'security/utils';
import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';

import PhoneNumberField from '../index';

jest.mock('hooks/useConfig');
jest.mock('hooks/useFlags');
jest.mock('hooks/useModal');
jest.mock('hooks/useThunkDispatch');
jest.mock('security/utils');

const mockAuth = userIsAuthenticated as jest.MockedFunction<typeof userIsAuthenticated>;
const mockConfig = useConfig as jest.MockedFunction<typeof useConfig>;
const mockFlags = useFlags as jest.MockedFunction<typeof useFlags>;
const mockModal = useModal as jest.MockedFunction<typeof useModal>;
const mockThunkDispatch = useThunkDispatch as jest.MockedFunction<typeof useThunkDispatch>;

const serviceDate = '3000-01-01';

const setupMocks = (isAuthenticated: boolean): void => {

  mockAuth.mockReturnValue(isAuthenticated);
  mockConfig.mockReturnValue({ ...testConfig });
  mockFlags.mockReturnValue({
    canChangeLanguage: true,
    developerMode: true,
  });
  mockModal.mockReturnValue({
    hideModal: jest.fn(),
    showModal: jest.fn(),
  });
  mockThunkDispatch.mockReturnValue(jest.fn());
};

const expectComponentToMatchSnapshot = (phoneNumber: string | null, language: string, isAuthenticated: boolean): void => {
  setupMocks(isAuthenticated);

  const tree = createSnapshotWithIntl(<PhoneNumberField phoneNumber={phoneNumber}
                                                        serviceDate={serviceDate} />, language);

  expect(tree).toMatchSnapshot();
};

const shouldNotLogErrorsInTheConsole = (phoneNumber: string | null, language: string, isAuthenticated: boolean): void => {
  setupMocks(isAuthenticated);

  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(<PhoneNumberField phoneNumber={phoneNumber} serviceDate={serviceDate} />, language);

  expect(spy).not.toHaveBeenCalled();
};

describe('<PhoneNumberField />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {

    it(`should not log errors in console when the current user is authenticated (${language})`, (): void =>
      shouldNotLogErrorsInTheConsole('7348881234', language, true));

    it(`should not log errors in console when the current user is not authenticated (${language})`, (): void =>
      shouldNotLogErrorsInTheConsole('7348881234', language, false));

    it(`should render correctly when a valid phone number is passed in and the current user is authenticated (${language})`, (): void =>
      expectComponentToMatchSnapshot('7348881234', language, true));

    // tslint:disable-next-line: max-line-length
    it(`should render correctly when a valid phone number is passed in and the current user is not authenticated (${language})`, (): void =>
      expectComponentToMatchSnapshot('7348881234', language, false));

    // tslint:disable-next-line: max-line-length
    it(`should hide the phone number edit link when the phone number is null, the current user is authenticated (${language})`, (): void =>
      expectComponentToMatchSnapshot(null, language, true));
  });
});
