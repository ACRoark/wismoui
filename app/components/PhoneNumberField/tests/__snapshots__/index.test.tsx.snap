// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneNumberField /> should hide the phone number edit link when the phone number is null, the current user is authenticated (en) 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Phone Number
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <div
      className="dte-wismo-phone-number-field"
    />
  </span>
</div>
`;

exports[`<PhoneNumberField /> should hide the phone number edit link when the phone number is null, the current user is authenticated (es) 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Número de teléfono
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <div
      className="dte-wismo-phone-number-field"
    />
  </span>
</div>
`;

exports[`<PhoneNumberField /> should render correctly when a valid phone number is passed in and the current user is authenticated (en) 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Phone Number
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <div
      className="dte-wismo-phone-number-field"
    >
      <span
        className="dte-wismo-phone-number-field-value"
      >
        (734) 888-1234
      </span>
      <div
        className="dte-wismo-edit-phone-number-link"
      >
        <button
          className="dte-wismo-button dte-wismo-button-link"
          data-track="mimo"
          data-track-action="click"
          data-track-detail="order tracker"
          onClick={[Function]}
        >
          Edit
        </button>
      </div>
    </div>
  </span>
</div>
`;

exports[`<PhoneNumberField /> should render correctly when a valid phone number is passed in and the current user is authenticated (es) 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Número de teléfono
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <div
      className="dte-wismo-phone-number-field"
    >
      <span
        className="dte-wismo-phone-number-field-value"
      >
        (734) 888-1234
      </span>
      <div
        className="dte-wismo-edit-phone-number-link"
      >
        <button
          className="dte-wismo-button dte-wismo-button-link"
          data-track="mimo"
          data-track-action="click"
          data-track-detail="order tracker"
          onClick={[Function]}
        >
          Editar
        </button>
      </div>
    </div>
  </span>
</div>
`;

exports[`<PhoneNumberField /> should render correctly when a valid phone number is passed in and the current user is not authenticated (en) 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Phone Number
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <div
      className="dte-wismo-phone-number-field"
    >
      <span
        className="dte-wismo-phone-number-field-value"
      >
        (734) 888-1234
      </span>
    </div>
  </span>
</div>
`;

exports[`<PhoneNumberField /> should render correctly when a valid phone number is passed in and the current user is not authenticated (es) 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Número de teléfono
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <div
      className="dte-wismo-phone-number-field"
    >
      <span
        className="dte-wismo-phone-number-field-value"
      >
        (734) 888-1234
      </span>
    </div>
  </span>
</div>
`;
